client/src/views/devices/ExportRecoveryDevicePage.vue

```diff
@@ -0,0 +1,208 @@
+                v-show="false"
+              />
+              <div v-if="recoveryKeyDownloaded && recoveryFileDownloaded">
+                <ion-button
+                  class="return-btn button-outline"
+                  @click="routerNavigateTo('workspaces')"
+                >
+                  <ion-icon
+                    :icon="home"
+                    class="icon"
+                  />
+                  {{ $t('ExportRecoveryDevicePage.actions.backToWorkspaces') }}
+                </ion-button>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </ion-content>
+  </ion-page>
+</template>
+
+<script setup lang="ts">
+import {
+  IonPage,
+  IonContent,
+  IonButton,
+} from '@ionic/vue';
+import MsInformativeText from '@/components/core/ms-text/MsInformativeText.vue';
+import { ref, inject } from 'vue';
+import { getPasswordFromUser } from '@/components/core/ms-modal/MsPasswordInputModal.vue';
+import { useI18n } from 'vue-i18n';
+import { exportRecoveryDevice, RecoveryDeviceErrorTag } from '@/parsec';
+import { getClientInfo } from '@/parsec/login';
+import { NotificationManager, Notification, NotificationKey, NotificationLevel } from '@/services/notificationManager';
+import { routerNavigateTo } from '@/router';
+import { home, checkmarkCircle } from 'ionicons/icons';
+
+const { t } = useI18n();
+
+enum ExportDevicePageState {
+  Start = 'start',
+  Download = 'download',
+}
+
+const state = ref(ExportDevicePageState.Start);
+let code = '';
+let file = '';
+const downloadLink = ref();
+const recoveryKeyDownloaded = ref(false);
+const recoveryFileDownloaded = ref(false);
+const notificationManager: NotificationManager = inject(NotificationKey)!;
+
+async function exportDevice(): Promise<void> {
+  const password = await getPasswordFromUser(t);
+  if (!password) {
+    return;
+  }
+  const result = await exportRecoveryDevice(password);
+  if (!result.ok) {
+    const notificationMsg = result.error.tag === RecoveryDeviceErrorTag.Invalid ?
+      t('PasswordInputModal.invalid') :
+      t('PasswordInputModal.otherError');
+    notificationManager.showToast(
+      new Notification({
+        message: notificationMsg,
+        level: NotificationLevel.Error,
+      }),
+    );
+    return;
+  }
+  code = result.value.code;
+  file = result.value.file;
+  state.value = ExportDevicePageState.Download;
+}
+
+async function downloadRecoveryKey(): Promise<void> {
+  const clientInfo = await getClientInfo();
+  fileDownload(code, t('ExportRecoveryDevicePage.filenames.recoveryKey', { org: (clientInfo.ok ? clientInfo.value.organizationId : '') }));
+  recoveryKeyDownloaded.value = true;
+  notificationManager.showToast(
+    new Notification({
+      message: t('ExportRecoveryDevicePage.toasts.keyDownloadOk'),
+      level: NotificationLevel.Success,
+    }),
+  );
+}
+
+async function downloadRecoveryFile(): Promise<void> {
+  const clientInfo = await getClientInfo();
```

Comment:
A bit of a waste to call `getClientInfo()` multiple times, it's better to do it once in the `onMounted()` function and store the result.